### PR TITLE
Identity: Remove typed id

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
We wan't to remote `TypedID` in favor of the string variant to make it consistent between uid:s and id:s.

With this I could also remove the internal `IsIdentityType` in favor of the same functionality exported from authlib. I also added a `IsIdenittyType` to Requester interface so it is more convenient to use.

Part of https://github.com/grafana/identity-access-team/issues/808

Sorry for the big pr but was hard to avoid

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/apimachinery/identity/requester.go [hunk 1]
```diff
@@ -15,13 +15,15 @@ type Requester interface {
...
+	IsIdentityType(expected ...claims.IdentityType) bool
 	// GetRawIdentifier returns only the identifier part of the UID, excluding the type
 	GetRawIdentifier() string
-	// Deprecated: use GetUID instead
+	// GetInternalID returns only the identifier part of the ID, excluding the type
 	GetInternalID() (int64, error)
 	// GetID returns namespaced internalID for the entity
 	// Deprecated: use GetUID instead
-	GetID() TypedID
+	GetID() string
```
Comment (ryantxu) [inline pkg/apimachinery/identity/requester.go:26]: do we need this one?    can use use GetInternalID or GetIdentityType instead?

File: pkg/services/authn/identity.go [hunk 2]
```diff
@@ -21,10 +22,11 @@ var _ identity.Requester = (*Identity)(nil)
 
 type Identity struct {
 	// ID is the unique identifier for the entity in the Grafana database.
-	// If the entity is not found in the DB or this entity is non-persistent, this field will be empty.
-	ID identity.TypedID
+	ID string
```
Comment (ryantxu) [inline pkg/services/authn/identity.go:25]:
```suggestion
	ID int64
```
is there any non-numeric case where we use ID?

File: pkg/apimachinery/identity/requester.go (hunk 1 above)
Comment (kalleep) [reply pkg/apimachinery/identity/requester.go:26]: I think we can, but I would rather do that in a follow up pr, there are many call sites to update to get rid of it.

File: pkg/services/accesscontrol/api/api.go [hunk 3]
```diff
@@ -72,23 +71,17 @@ func (api *AccessControlAPI) searchUsersPermissions(c *contextmodel.ReqContext)
 		ActionPrefix: c.Query("actionPrefix"),
 		Action:       c.Query("action"),
 		Scope:        c.Query("scope"),
+		TypedID:      c.Query("namespacedId"),
```
[the remark this answers is not in the document]
Comment (kalleep) [reply pkg/services/accesscontrol/api/api.go:74]: Yes this is still used and yeah that status code was probably a typo

**Review by ryantxu — APPROVED** [review]
good progress -- would still love to see how we can get rid of the `GetID()` duplicaiton/confusion

**Comment by kalleep** [comment]
> good progress -- would still love to see how we can get rid of the `GetID()` duplicaiton/confusion

Yes I will look into that. Also there are other functions on the interface we can remove e.g. GetDisplayName is a duplicate of GetName from k8s user.Info interface. I will continue to remove / simplify this one

**Pull request merged**